src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.Tests/CallSiteTests.cs

```diff
@@ -361,6 +388,13 @@ public void Dispose()
             }
         }
 
+        private static T GetServiceFromCallSiteFactory<T>(ServiceProvider provider)
+        {
+            var callSite = provider.CallSiteFactory.GetCallSite(typeof(T), new CallSiteChain());
```

Comment:
Can we write this test in terms of the public API? I don't think we should be using `internal` APIs unless we absolutely must.

Reply:
Done :) The test even became simpler. Thanks